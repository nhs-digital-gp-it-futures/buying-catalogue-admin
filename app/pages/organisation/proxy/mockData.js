export const organisation = {
  organisationId: 'b7ee5261-43e7-4589-907b-5eef5e98c085',
  name: 'Cheshire and Merseyside Commissioning Hub',
  odsCode: '13Y',
  primaryRoleId: 'RO98',
  address: {
    line1: 'C/O NHS ENGLAND, 1W09, 1ST FLOOR',
    line2: 'QUARRY HOUSE',
    line3: 'QUARRY HILL',
    town: 'LEEDS',
    county: 'WEST YORKSHIRE',
    postcode: 'LS2 7UE',
    country: 'ENGLAND',
  },
  catalogueAgreementSigned: false,
};

export const unrelatedOrgs = [
  {
    organisationId: '43cc8258-6e92-4d39-9296-edc5bb000563',
    name: 'Cumbria and North East Commissioning Hub',
    odsCode: '13X',
  },
  {
    organisationId: '6d37d7fb-df06-403c-9c46-d649919f9158',
    name: 'East – H&J Commissioning Hub',
    odsCode: '14R',
  },
  {
    organisationId: 'bb21dd07-58df-457a-a9cb-6b035684692b',
    name: 'East Commissioning Hub',
    odsCode: '14E',
  },
  {
    organisationId: 'd795a598-26f0-46fb-8a6e-f22cd912d7dc',
    name: 'Greater Manchester Commissioning Hub',
    odsCode: '14J',
  },
  {
    organisationId: 'dff57334-4055-48cf-a569-da490ccdc66a',
    name: 'Hampshire, Isle of Wight and Thames Valley – H&J Commissioning Hub',
    odsCode: '14W',
  },
];

export const unrelatedOrgsRadio = [
  {
    value: '43cc8258-6e92-4d39-9296-edc5bb000563',
    text: 'Cumbria and North East Commissioning Hub',
  },
  {
    value: '6d37d7fb-df06-403c-9c46-d649919f9158',
    text: 'East – H&J Commissioning Hub',
  },
  {
    value: 'bb21dd07-58df-457a-a9cb-6b035684692b',
    text: 'East Commissioning Hub',
  },
  {
    value: 'd795a598-26f0-46fb-8a6e-f22cd912d7dc',
    text: 'Greater Manchester Commissioning Hub',
  },
  {
    value: 'dff57334-4055-48cf-a569-da490ccdc66a',
    text: 'Hampshire, Isle of Wight and Thames Valley – H&J Commissioning Hub',
  },
];

export const relatedOrgs = [
  {
    organisationId: '2459261a-d8f0-464a-bff7-45b2d7adb1ef',
    name: 'Yorkshire and Humber Commissioning Hub',
    odsCode: '13V',
  },
  {
    organisationId: '7a9b95b0-1d0b-4ba4-84f6-59d710938141',
    name: 'South East – H&J Commissioning Hub',
    odsCode: '97T',
  },
  {
    organisationId: '6672cfe8-1583-4d02-a0af-83d29bcb0513',
    name: 'South West South Commissioning Hub',
    odsCode: '15G',
  },
  {
    organisationId: 'bac2e2e0-143f-412a-9e78-8618f09cf779',
    name: 'West Midlands Commissioning Hub',
    odsCode: '14C',
  },
  {
    organisationId: 'a98f01d6-3f73-4b59-b213-e2bd3397c16f',
    name: 'Cumbria and North East – H&J Commissioning Hub',
    odsCode: '14P',
  },
];

export const relatedOrgsTable = [
  [
    {
      data: 'Yorkshire and Humber Commissioning Hub',
      dataTestId: 'related-org-name-2459261a-d8f0-464a-bff7-45b2d7adb1ef',
    },
    {
      data: '13V',
      dataTestId: 'related-org-odsCode-2459261a-d8f0-464a-bff7-45b2d7adb1ef',
    },
    {
      href: '/admin/organisations/removeproxy/orgId/2459261a-d8f0-464a-bff7-45b2d7adb1ef',
      data: 'Remove',
      dataTestId: 'related-org-remove-2459261a-d8f0-464a-bff7-45b2d7adb1ef',
    },
  ],
  [
    {
      data: 'South East – H&J Commissioning Hub',
      dataTestId: 'related-org-name-7a9b95b0-1d0b-4ba4-84f6-59d710938141',
    },
    {
      data: '97T',
      dataTestId: 'related-org-odsCode-7a9b95b0-1d0b-4ba4-84f6-59d710938141',
    },
    {
      href: '/admin/organisations/removeproxy/orgId/7a9b95b0-1d0b-4ba4-84f6-59d710938141',
      data: 'Remove',
      dataTestId: 'related-org-remove-7a9b95b0-1d0b-4ba4-84f6-59d710938141',
    },
  ],
  [
    {
      data: 'South West South Commissioning Hub',
      dataTestId: 'related-org-name-6672cfe8-1583-4d02-a0af-83d29bcb0513',
    },
    {
      data: '15G',
      dataTestId: 'related-org-odsCode-6672cfe8-1583-4d02-a0af-83d29bcb0513',
    },
    {
      href: '/admin/organisations/removeproxy/orgId/6672cfe8-1583-4d02-a0af-83d29bcb0513',
      data: 'Remove',
      dataTestId: 'related-org-remove-6672cfe8-1583-4d02-a0af-83d29bcb0513',
    },
  ],
  [
    {
      data: 'West Midlands Commissioning Hub',
      dataTestId: 'related-org-name-bac2e2e0-143f-412a-9e78-8618f09cf779',
    },
    {
      data: '14C',
      dataTestId: 'related-org-odsCode-bac2e2e0-143f-412a-9e78-8618f09cf779',
    },
    {
      href: '/admin/organisations/removeproxy/orgId/bac2e2e0-143f-412a-9e78-8618f09cf779',
      data: 'Remove',
      dataTestId: 'related-org-remove-bac2e2e0-143f-412a-9e78-8618f09cf779',
    },
  ],
  [
    {
      data: 'Cumbria and North East – H&J Commissioning Hub',
      dataTestId: 'related-org-name-a98f01d6-3f73-4b59-b213-e2bd3397c16f',
    },
    {
      data: '14P',
      dataTestId: 'related-org-odsCode-a98f01d6-3f73-4b59-b213-e2bd3397c16f',
    },
    {
      href: '/admin/organisations/removeproxy/orgId/a98f01d6-3f73-4b59-b213-e2bd3397c16f',
      data: 'Remove',
      dataTestId: 'related-org-remove-a98f01d6-3f73-4b59-b213-e2bd3397c16f',
    },
  ],
];

export const errors = {
  messagge: 'Select the organisation you are looking for',
  errorList: [{
    text: 'Select the organisation you are looking for',
    href: '#relatedOrganisationId',
  }],
};
