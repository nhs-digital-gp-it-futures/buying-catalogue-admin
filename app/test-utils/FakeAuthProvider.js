export class FakeAuthProvider {
  // eslint-disable-next-line no-useless-constructor, no-empty-function
  constructor(fakeLogoutMethod = () => {}) {
    this.fakeLogout = fakeLogoutMethod;
  }

  // eslint-disable-next-line class-methods-use-this
  setup(app) {
    app.use((req, res, next) => {
      if (req.cookies && req.cookies.fakeToken) {
        req.user = JSON.parse(req.cookies.fakeToken);
      }
      req.logout = this.fakeLogout;
      next();
    });
  }

  // eslint-disable-next-line class-methods-use-this, no-unused-vars
  login() {
    return (req, res) => {
      res.redirect('http://identity-server/login');
    };
  }

  // eslint-disable-next-line class-methods-use-this, no-unused-vars
  loginCallback() {
    return (req, res) => {
      res.redirect('/');
    };
  }

  // eslint-disable-next-line class-methods-use-this
  logout() {
    return '/signout-callback-oidc';
  }

  // eslint-disable-next-line class-methods-use-this
  authorise() {
    return (req, res, next) => {
      if (!req.user) {
        this.login()(req, res, next);
      } else if (req.user && req.user.organisation) {
        next();
      } else {
        throw new Error('Not authorised matey');
      }
    };
  }
}
