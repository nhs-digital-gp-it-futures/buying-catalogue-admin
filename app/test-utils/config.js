export const apiLocalhost = 'http://localhost:5100';

export const organisationsApiLocalhost = 'http://localhost:8075';

export const apiPath = '/api/v1';

export const clientLocalhost = 'http://localhost:1234';
