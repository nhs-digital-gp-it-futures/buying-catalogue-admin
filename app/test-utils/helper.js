import cheerio from 'cheerio';
import request from 'supertest';

export const extractInnerText = async (element) => {
  const elementInnerText = await element.innerText;
  return elementInnerText.trim();
};

const extractCsrfToken = (res) => {
  const $ = cheerio.load(res.text);
  return $('[name=_csrf]').val();
};

export const getCsrfTokenFromGet = async (app, getPath, authenticationCookie) => {
  let cookies;
  let csrfToken;

  await request(app)
    .get(getPath)
    .set('Cookie', authenticationCookie)
    .then((getRes) => {
      cookies = getRes.headers['set-cookie'];
      csrfToken = extractCsrfToken(getRes);
    });
  return { cookies, csrfToken };
};

export const setFakeCookie = async (app, getPath) => {
  let cookies;
  await request(app)
    .get(getPath)
    .set('Cookie', ['cookie1=cookie1value'])
    .then((getRes) => {
      cookies = getRes.headers['set-cookie'];
    });

  return { modifiedApp: app, cookies };
};
