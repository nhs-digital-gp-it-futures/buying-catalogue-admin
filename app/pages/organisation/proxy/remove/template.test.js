import { componentTester, snapshotTest } from '../../../../test-utils/componentTester';
import { organisation } from '../mockData';

const setup = {
  template: {
    path: 'pages/organisation/proxy/remove/template.njk',
  },
};

describe('add proxy for organisation', () => {
  it('the page should render', componentTester(setup, (harness) => {
    harness.request({ organisation }, ($) => {
      const snapshot = snapshotTest($, '[data-test-id="main-content"]');
      expect(snapshot).toMatchSnapshot();
    });
  }));
});
