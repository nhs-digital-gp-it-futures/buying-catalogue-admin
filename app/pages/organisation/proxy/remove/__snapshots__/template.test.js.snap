// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add proxy for organisation the page should render 1`] = `
"<main class=\\"nhsuk-main-wrapper\\" id=\\"maincontent\\" data-test-id=\\"main-content\\">
        
  <div class=\\"nhsuk-width-container\\" data-test-id=\\"remove-proxy-organisation-page\\">
    <div data-test-id=\\"go-back-link\\">
      <div class=\\"nhsuk-back-link\\">
  <a class=\\"nhsuk-back-link__link\\" href=\\"backLinkUrl\\">
  <svg class=\\"nhsuk-icon nhsuk-icon__chevron-left\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"24\\" width=\\"24\\">
    <path d=\\"M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z\\"></path>
  </svg>
  Go back</a>
</div>

    </div>

    <div class=\\"nhsuk-grid-row\\">
      <div class=\\"nhsuk-grid-column-two-thirds\\">
        <h1 data-test-id=\\"org-page-title\\" class=\\"nhsuk-u-margin-top-7 nhsuk-u-margin-bottom-3\\">Remove organisation</h1>
        <p data-test-id=\\"organisation-details-subheading\\" class=\\"nhsuk-heading-l nhsuk-u-font-size-22 nhsuk-u-margin-bottom-7\\">
          This organisation will no longer be able to act on behalf of Cheshire and Merseyside Commissioning Hub.
        </p>
        <form method=\\"POST\\">
          <input type=\\"hidden\\" name=\\"_csrf\\" value=\\"\\">
          

  
  

  
  

  <div data-test-id=\\"submit-button\\">
    


  
    
  




<button class=\\"nhsuk-button\\" type=\\"submit\\">
  Remove organisation
</button>



  </div>

          <a href=\\"backLinkUrl\\">Cancel</a>
        </form>
      </div>
    </div>
  </div>

      </main>"
`;
